#!/usr/bin/env python

PACKAGE="dynamic_reconfig"

from dynamic_reconfigure.parameter_generator_catkin import *

gen=ParameterGenerator()

gen.add("int_a",int_t,0,"An integer parameter",50,-100,100)
gen.add("double_a",double_t,0,"A double parameter",50.0,-100.0,100.0)
gen.add("bool_a", bool_t, 0, "bool parameter", True);
gen.add("str_a",str_t,0,"A str parameter","hello world")
size_enum=gen.enum([gen.const("small",int_t,0,"a small constant"),gen.const("medium",int_t,1,"a medium constant"),gen.const("big",int_t,2,"a big constant")],"select from the list")
gen.add("size",int_t,0,"a size parameter",1,0,2,edit_method=size_enum)

exit(gen.generate(PACKAGE,"dynamic_reconfig","dynamic_"))
